### ISSUE TITLE
problem with readers.las

### ISSUE THREAD

**Issue by epifanio** [chambbj/learning-pdal#1]
Hi, thanks for making this tutorial available!
I'm trying to run the notebooks on  osx, PDAL built from src (github master branch)

Running the `Getting Started` notebook, I got an error in reading the sample data file.
The error appear when tring to validate the json file, at:

```
p = pdal.Pipeline(json)
p.validate() 
```
The  log says:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-19-66e41e4f1c18> in <module>()
      1 p = pdal.Pipeline(json)
----> 2 p.validate() # check if our JSON and options were good
      3 p.loglevel = 8 #really noisy
      4 count = p.execute()
      5 data = p.arrays[0]

~/.virtualenvs/jupyter/lib/python3.6/site-packages/PDAL-1.5.0-py3.6-macosx-10.13-x86_64.egg/pdal/pipeline.py in validate(self)
     40 
     41     def validate(self):
---> 42         return self.p.validate()
     43 
     44     def get_arrays(self):

libpdalpython.pyx in pdal.libpdalpython.PyPipeline.validate()

RuntimeError: readers.las: File signature is not 'LASF', is this an LAS/LAZ file?
```
 I tried to check if the path to the file was correct, this the notebook I used:

https://gist.github.com/598a11f456cfad28d555c3e23af738e0

Thanks for any pointers to solve this!

**Comment by epifanio** [comment]
Seems I found the problem .. 
I generated the data directory using the following commands:

```
cd notebooks/data
git submodule init
git submodule update
```

which gave me a laz file of ~130 bytes.
Suspicious size ... browsing the repository, I downloaded directly the [file](
https://github.com/PDAL/data/blob/3bd0322fae9fd60ec7874b5ed8bfdd9356a6cfb6/isprs/samp11-utm.laz) (~100 KB) and everything works as expected.

I guess the git submodule commands were not enough to generate the data.

**Comment by chambbj** [comment]
Sorry to hear that it's not working for you. I attempted to recreate your issue, but was unable.

My steps:

```
$ git clone https://github.com/chambbj/learning-pdal
Cloning into 'learning-pdal'...
remote: Counting objects: 54, done.
remote: Total 54 (delta 0), reused 0 (delta 0), pack-reused 54
Unpacking objects: 100% (54/54), done.
Checking connectivity... done.
$ cd learning-pdal
$ git submodule init
Submodule 'notebooks/data' (https://github.com/pdal/data) registered for path 'notebooks/data'
$ git submodule update
Cloning into 'notebooks/data'...
remote: Counting objects: 73, done.
remote: Total 73 (delta 0), reused 0 (delta 0), pack-reused 73
Unpacking objects: 100% (73/73), done.
Checking connectivity... done.
Downloading autzen-utm-chipped-25-v1.bpf (46.30 KB)
Downloading autzen-utm-chipped-25-v3-deflate-interleaved.bpf (23.54 KB)
Downloading autzen-utm-chipped-25-v3-deflate-segregated.bpf (18.41 KB)
Downloading autzen-utm-chipped-25-v3-deflate.bpf (19.34 KB)
Downloading autzen-utm-chipped-25-v3-fastlz-interleaved.bpf (28.66 KB)
Downloading autzen-utm-chipped-25-v3-fastlz-segregated.bpf (21.09 KB)
Downloading autzen-utm-chipped-25-v3-fastlz.bpf (26.34 KB)
Downloading autzen-utm-chipped-25-v3-interleaved.bpf (46.54 KB)
Downloading autzen-utm-chipped-25-v3-segregated.bpf (46.54 KB)
Downloading autzen-utm-chipped-25-v3.bpf (46.54 KB)
Downloading isprs/CSite1_orig-utm.laz (4.33 MB)
Downloading isprs/CSite1_red1-utm.laz (1.83 MB)
Downloading isprs/CSite1_red2-utm.laz (536.91 KB)
Downloading isprs/CSite2_orig-utm.laz (1.40 MB)
Downloading isprs/CSite3_orig-utm.laz (1.15 MB)
Downloading isprs/CSite4_orig-utm.laz (1.46 MB)
Downloading isprs/FSite5_orig-utm.laz (1.97 MB)
Downloading isprs/FSite6_orig-utm.laz (1.62 MB)
Downloading isprs/FSite7_orig-utm.laz (1.10 MB)
Downloading isprs/FSite8_orig-utm.laz (1013.50 KB)
Downloading isprs/FSite8_red1-utm.laz (280.43 KB)
Downloading isprs/FSite8_red2-utm.laz (79.67 KB)
Downloading isprs/samp11-utm.laz (97.23 KB)
Downloading isprs/samp12-utm.laz (120.08 KB)
Downloading isprs/samp21-utm.laz (28.32 KB)
Downloading isprs/samp22-utm.laz (64.12 KB)
Downloading isprs/samp23-utm.laz (55.78 KB)
Downloading isprs/samp24-utm.laz (17.27 KB)
Downloading isprs/samp31-utm.laz (62.89 KB)
Downloading isprs/samp41-utm.laz (22.59 KB)
Downloading isprs/samp42-utm.laz (81.12 KB)
Downloading isprs/samp51-utm.laz (42.57 KB)
Downloading isprs/samp52-utm.laz (58.10 KB)
Downloading isprs/samp53-utm.laz (83.20 KB)
Downloading isprs/samp54-utm.laz (26.33 KB)
Downloading isprs/samp61-utm.laz (75.42 KB)
Downloading isprs/samp71-utm.laz (36.75 KB)
Downloading workshop/18TWK820985.laz (32.34 MB)
Downloading workshop/S1C1_csd_004.csd (350.29 MB)
Downloading workshop/autzen.laz (53.74 MB)
Downloading workshop/csite-dd.laz (4.40 MB)
Downloading workshop/interesting.las (36.81 KB)
Downloading workshop/interesting.laz (18.35 KB)
Downloading workshop/uncompahgre.laz (78.30 MB)
Submodule path 'notebooks/data': checked out '3bd0322fae9fd60ec7874b5ed8bfdd9356a6cfb6'
```

A `--recursive` clone also worked:

```
$ git clone --recursive https://github.com/chambbj/learning-pdal
Cloning into 'learning-pdal'...
remote: Counting objects: 54, done.
remote: Total 54 (delta 0), reused 0 (delta 0), pack-reused 54
Unpacking objects: 100% (54/54), done.
Checking connectivity... done.
Submodule 'notebooks/data' (https://github.com/pdal/data) registered for path 'notebooks/data'
Cloning into 'notebooks/data'...
remote: Counting objects: 73, done.
remote: Total 73 (delta 0), reused 0 (delta 0), pack-reused 73
Unpacking objects: 100% (73/73), done.
Checking connectivity... done.
Downloading autzen-utm-chipped-25-v1.bpf (46.30 KB)
Downloading autzen-utm-chipped-25-v3-deflate-interleaved.bpf (23.54 KB)
Downloading autzen-utm-chipped-25-v3-deflate-segregated.bpf (18.41 KB)
Downloading autzen-utm-chipped-25-v3-deflate.bpf (19.34 KB)
Downloading autzen-utm-chipped-25-v3-fastlz-interleaved.bpf (28.66 KB)
Downloading autzen-utm-chipped-25-v3-fastlz-segregated.bpf (21.09 KB)
Downloading autzen-utm-chipped-25-v3-fastlz.bpf (26.34 KB)
Downloading autzen-utm-chipped-25-v3-interleaved.bpf (46.54 KB)
Downloading autzen-utm-chipped-25-v3-segregated.bpf (46.54 KB)
Downloading autzen-utm-chipped-25-v3.bpf (46.54 KB)
Downloading isprs/CSite1_orig-utm.laz (4.33 MB)
Downloading isprs/CSite1_red1-utm.laz (1.83 MB)
Downloading isprs/CSite1_red2-utm.laz (536.91 KB)
Downloading isprs/CSite2_orig-utm.laz (1.40 MB)
Downloading isprs/CSite3_orig-utm.laz (1.15 MB)
Downloading isprs/CSite4_orig-utm.laz (1.46 MB)
Downloading isprs/FSite5_orig-utm.laz (1.97 MB)
Downloading isprs/FSite6_orig-utm.laz (1.62 MB)
Downloading isprs/FSite7_orig-utm.laz (1.10 MB)
Downloading isprs/FSite8_orig-utm.laz (1013.50 KB)
Downloading isprs/FSite8_red1-utm.laz (280.43 KB)
Downloading isprs/FSite8_red2-utm.laz (79.67 KB)
Downloading isprs/samp11-utm.laz (97.23 KB)
Downloading isprs/samp12-utm.laz (120.08 KB)
Downloading isprs/samp21-utm.laz (28.32 KB)
Downloading isprs/samp22-utm.laz (64.12 KB)
Downloading isprs/samp23-utm.laz (55.78 KB)
Downloading isprs/samp24-utm.laz (17.27 KB)
Downloading isprs/samp31-utm.laz (62.89 KB)
Downloading isprs/samp41-utm.laz (22.59 KB)
Downloading isprs/samp42-utm.laz (81.12 KB)
Downloading isprs/samp51-utm.laz (42.57 KB)
Downloading isprs/samp52-utm.laz (58.10 KB)
Downloading isprs/samp53-utm.laz (83.20 KB)
Downloading isprs/samp54-utm.laz (26.33 KB)
Downloading isprs/samp61-utm.laz (75.42 KB)
Downloading isprs/samp71-utm.laz (36.75 KB)
Downloading workshop/18TWK820985.laz (32.34 MB)
Downloading workshop/S1C1_csd_004.csd (350.29 MB)
Downloading workshop/autzen.laz (53.74 MB)
Downloading workshop/csite-dd.laz (4.40 MB)
Downloading workshop/interesting.las (36.81 KB)
Downloading workshop/interesting.laz (18.35 KB)
Downloading workshop/uncompahgre.laz (78.30 MB)
Submodule path 'notebooks/data': checked out '3bd0322fae9fd60ec7874b5ed8bfdd9356a6cfb6'
```

Any more information you can provide to help troubleshoot? Git version? OS X version?

**Comment by epifanio** [comment]
From a fresh attempt:

```
Massimos-MacBook-Pro:~ epi$ cd dev/
Massimos-MacBook-Pro:dev epi$ mkdir test_pdal_workshop
Massimos-MacBook-Pro:dev epi$ cd test_pdal_workshop/
Massimos-MacBook-Pro:test_pdal_workshop epi$ git clone https://github.com/chambbj/learning-pdal
Cloning into 'learning-pdal'...
remote: Counting objects: 54, done.
remote: Total 54 (delta 0), reused 0 (delta 0), pack-reused 54
Unpacking objects: 100% (54/54), done.
Massimos-MacBook-Pro:test_pdal_workshop epi$ cd learning-pdal/notebooks/data
Massimos-MacBook-Pro:data epi$ git submodule init
Submodule 'notebooks/data' (https://github.com/pdal/data) registered for path './'
Massimos-MacBook-Pro:data epi$ git submodule update
Cloning into '/Users/epi/dev/test_pdal_workshop/learning-pdal/notebooks/data'...
Submodule path './': checked out '3bd0322fae9fd60ec7874b5ed8bfdd9356a6cfb6'
Massimos-MacBook-Pro:data epi$ 
```
This is on `OSX 10.13` running `git version 2.13.5 (Apple Git-94)`
I tried the same steps on a linux machine `Debian 4.12.13-1 (2017-09-19) x86_64 GNU/Linux` running `git version 2.14.2` with the same results. I wonder if it is a branch issue.

Note: an attempt to clone the PDAL/Data repository directly gave the same problem:
```
epinux@geohab-tools:~/dev$ git clone https://github.com/PDAL/data
Cloning into 'data'...
remote: Counting objects: 73, done.
remote: Total 73 (delta 0), reused 0 (delta 0), pack-reused 73
Unpacking objects: 100% (73/73), done.

epinux@geohab-tools:~/dev$ ls -la data
total 68
drwxr-xr-x  5 epinux epinux 4096 Sep 30 12:47 .
drwxr-xr-x 26 epinux epinux 4096 Sep 30 12:47 ..
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v1.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-deflate.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-deflate-interleaved.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-deflate-segregated.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-fastlz.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-fastlz-interleaved.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-fastlz-segregated.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-interleaved.bpf
-rw-r--r--  1 epinux epinux  130 Sep 30 12:47 autzen-utm-chipped-25-v3-segregated.bpf
drwxr-xr-x  8 epinux epinux 4096 Sep 30 12:47 .git
-rw-r--r--  1 epinux epinux  168 Sep 30 12:47 .gitattributes
drwxr-xr-x  2 epinux epinux 4096 Sep 30 12:47 isprs
-rw-r--r--  1 epinux epinux   70 Sep 30 12:47 README.md
drwxr-xr-x  2 epinux epinux 4096 Sep 30 12:47 workshop
```

**Comment by chambbj** [comment]
Do you have git lfs installed? If not, please see instructions [here](https://help.github.com/articles/installing-git-large-file-storage/). The data repository is an lfs repository, so I'm guessing without lfs installed, you are only getting the pointers.

**Comment by epifanio** [comment]
Thanks for pointing me to `git lfs` i didn't know about it. I installed and enableed the tool with `git lfs install` the tool on mac and the problem is gone. Thanks!

**Comment by chambbj** [comment]
Closing because you finally got it to work? Or just moving on? It is
baffling to me that `lfs` didn't do the trick.

And did you try `git clone --recursive
https://github.com/chambbj/learning-pdal`?

**Comment by epifanio** [comment]
I got it to work on OSX after initializating `lfs` and I used `git clone --recursive https://github.com/chambbj/learning-pdal`. I just now tried on my linux machine, but the problem persist there. So I confirm the `git lfs` installation and the use of `--recursive` did the trick on osx .. but not on linux. I suspect there is something i need to set on the linux machine to have it working. tring to look into it.
